api/external/nodes/manager/manager.proto

```diff
@@ -169,62 +246,141 @@ service NodeManagerService {
+	};
+	// UUID for the node manager.
 	string id = 1;
 }
 message Ids {
+	// List of node manager UUIDs.
 	repeated Id ids = 1;
 }
 
 message Query {
+	option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+		json_schema: {
+			required: ["filter_map"]
+			}
+		example: {value: '{"filter_map": [{"key": "manager_type","values":["aws-ec2"]}],"sort": "date_added"}'}
+	};
+	// Filters for the query: "manager_type".
 	repeated chef.automate.domain.compliance.api.common.Filter filter_map = 25;
-
+	// Sort the results in ascending or descending order.
 	enum OrderType {
 		ASC = 0;
 		DESC = 1;
 	}
 	OrderType order = 21;
+	// Field to use for sorting.
+	// Valid fields are: name, type, status, status_message, date_added.
 	string sort = 22;
+	// Starting page for the list. For example, if your query returns 100 pages,
+	// and you know you're looking for a node manager somewhere in the middle,
+	// you might want to start on 50.
 	int32 page = 23;
+	// Number of results on each page.
 	int32 per_page = 24;
 }
 
 message Fields {
+	option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+		json_schema: {
+			required: ["fields"]
+		}
+	};
+	// One or more fields: regions, tags, name, subscription_id.
 	repeated string fields = 1;
 }
 
 message FieldQuery {
-		Query query = 20;
-		string field = 21;
-		string node_manager_id = 22;
+	option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+		json_schema: {
+			required: ["node_manager_id"]
+		}
+	};
+	// Query details (filters) to be applied to the results.
+	Query query = 20;
+	// Possible search fields: regions, tags, name, subscription_id.
+	string field = 21;
+	// Node manager ID.
+	string node_manager_id = 22;
 }
 
 message Nodes {
-		repeated string nodes = 1;
-		int32 total = 20;
+	option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+		json_schema: {
+			required: ["nodes"]
+		}
+	};
+	// List of node names matching the request.
+	repeated string nodes = 1;
+	// Total count of node names matching the request.
+	int32 total = 20;
 }
 
 message NodeQuery {
-		Query query = 20;
-		string node_manager_id = 21;
+	option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+		json_schema: {
+			required: ["node_manager_id"]
+		}
+	};
+	// Valid search filters: manager_type.
+	Query query = 20;
+	// Node manager ID.
+	string node_manager_id = 21;
 }
 
 message NodeManager {
-		string id = 1;
-		string name = 2;
-		string type = 3;
-		string credential_id = 20; // aws or azure cred id
-		repeated CredentialsByTags instance_credentials = 21; // ssh, winrm creds
-		string status = 22;
-		string account_id = 24;
-		google.protobuf.Timestamp date_added = 25;
-		repeated chef.automate.domain.compliance.api.common.Kv credential_data = 26;
+	option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+		json_schema: {
+			required: ["id","credential_id","credential_data","type"]
+		}
```

Comment:
hmmm. id and type are required, and then we need *either* credential id or credential data. if we receive both, we will override the data in credential id with the data found in credential data

Reply:
Yah, I saw that in the code but we don't have a conditional case for marking things as required. We can 1) prefer one over the other and only use that in examples 2) mark both as required, use very clear language to explain the distinction, and only provide examples of one case or  3) mark neither as required but provide an example.